core/update.php

```diff
@@ -283,61 +285,120 @@ function update_results_page() {
+
+    $elements = backdrop_get_form('update_script_overview_form');
+    $output .= backdrop_render($elements);
+  }
+
+  // If this is an upgrade from Drupal 7, check if additional modules need to be
+  // enabled, and present them in a list.
   $module_status_report = update_upgrade_check_dependencies();
 	if (!empty($module_status_report)) {
     $output .= $module_status_report;
   }
-  $form_action = check_url(backdrop_current_script_url(array('op' => 'selection', 'token' => $token)));
-  $output .= '<form method="post" action="' . $form_action . '">
-  <div class="form-actions">
-    <input type="submit" value="Continue" class="form-submit button-primary" />
-    <a href="' . base_path() . '">Cancel</a>
-  </div>
-  </form>';
-  $output .= "\n";
+
   return $output;
 }
 
+/**
+ * Form constructor for the "Overview" step of update.php.
+ *
+ * This form is an intermediate step in the database/configuration update script
+ * workflow, that is shown after the "Verify requirements" step (see
+ * update_check_requirements()). The purpose of this page is to encourage taking
+ * a backup of the database and the site, and to give the opportunity to put the
+ * site in maintenance mode. After this step, the "Review updates" step follows
+ * (see update_selection_page()).
+ *
+ * @see update_script_overview_form_submit()
+ * @ingroup forms
+ */
+function update_script_overview_form($form, &$form_state) {
+  $maintenance_mode = state_get('maintenance_mode');
+  // Store the current maintenance mode status so we can restore it when done.
+  $_SESSION['maintenance_mode'] = $maintenance_mode;
+
+  if ($maintenance_mode) {
+    backdrop_set_message(t('The site is already in maintenance mode.'));
+  }
+
+  $form['maintenance_mode'] = array(
+    '#type' => 'checkbox',
+    '#id' => 'maintenance_mode',
+    '#title' => 'Perform updates with site in maintenance mode (<strong>strongly recommended</strong>)',
+    '#default_value' => TRUE,
+    '#disabled' => $maintenance_mode,
+  );
+
+  $form['actions'] = array('#type' => 'actions');
+  $form['actions']['submit'] = array(
+    '#type' => 'submit',
+    '#value' => 'Continue',
+  );
+  $form['actions']['cancel'] = array(
+    '#type' => 'link',
+    '#title' => 'Cancel',
+    '#href' => base_path(),
+  );
+
+  return $form;
+}
+
+/**
+ * Form submission handler for update_script_overview_form().
+ *
+ * If the site administrator selected the option to put the site in maintenance
+ * mode during the update, do so now. Then redirect to the "Review updates"
+ * step.
+ *
+ * @see update_script_overview_form()
+ */
+function update_script_overview_form_submit($form, &$form_state) {
+  global $base_url;
+
+  // Store maintenance_mode setting so we can restore it when done.
+  $_SESSION['maintenance_mode'] = config_get('system.core', 'maintenance_mode');
+  if ($form_state['values']['maintenance_mode'] == TRUE) {
+    state_set('maintenance_mode', TRUE);
+  }
+
```

Comment:
The code should set *maintenance_mode* to the value selected from users. If they do not want the site to be in maintenance mode, the site should not be in maintenance mode.